hibernate-core/src/main/java/org/hibernate/query/sqm/internal/ConcreteSqmSelectQueryPlan.java

```diff
@@ -159,12 +159,16 @@ private RowTransformer makeRowTransformerTupleTransformerAdapter(
 				sqmInterpretation.getTableGroupAccess()::findTableGroup,
 				session
 		);
-		sqmInterpretation.getJdbcSelect().bindFilterJdbcParameters( jdbcParameterBindings );
+
+		final JdbcSelect jdbcSelect = sqmInterpretation.getJdbcSelect();
+		jdbcSelect.bindFilterJdbcParameters( jdbcParameterBindings );
+
+		session.autoFlushIfRequired( jdbcSelect.getAffectedTableNames() );
```

Comment:
I already did that two lines below. One of the two must go :)

Reply:
Thanks @beikov :+1:,
  added it before merging your branch and haven't noticed you did the same.
Fixing it and then I'll merge it!